detekt-rules/src/main/java/com/wire/detekt/rules/EnforceSerializableFields.kt

```diff
@@ -42,12 +45,18 @@ class EnforceSerializableFields(config: Config = Config.empty) : Rule(config) {
         }
     }
 
-    private fun getDeclarationsOfInterestForClass(kClass: KtClassOrObject): List<KtAnnotated> =
-        if (kClass is KtClass && kClass.isEnum()) {
-            kClass.declarations.filterIsInstance<KtEnumEntry>()
+    private fun getDeclarationsOfInterestForClass(kClass: KtClassOrObject): List<KtAnnotated> {
+        return if (kClass is KtClass && kClass.isEnum()) {
+            if (kClass.annotationEntries.firstOrNull()?.getValueArgumentsInParentheses()?.isNotEmpty() == true) {
```

Comment:
```kotlin
@Suppress("MemberVisibilityCanBePrivate")
@Serializable
enum class Whatever {
    FOO, BAR;
}
```

I think `annotationEntries.firstOrNull()` would return the `Suppress`, and not the `Serializable`.

So this wouldn't be right.


Also, I think the logic for custom serializable should be applied for classes too, not only for enums.


So, if we're going this way:

```kotlin
private fun getDeclarationsOfInterestForClass(serializableAnnotation: KtAnnotationEntry, kClass: KtClassOrObject): List<KtAnnotated> {
    if (serializableAnnotation.getValueArgumentsInParentheses().isNotEmpty()) {
        println("Ignoring internal entries for [${kClass.name}] since it has a custom [KSerializer] implementation")
        return emptyList()
    }
    ... // Resume with old logic here.
}
    

// Usage:

getDeclarationsOfInterestForClass(superEntry, kClass)

```